fbgemm_gpu/hip_kernel/split_tbe_bwd.hip.hpp

```diff
@@ -0,0 +1,102 @@
+            uint32_t max_segment_length_per_warp,                                                                                    \
+            const float * p_indice_weights,                                                                                          \
+            uint32_t emb_dim,                                                                                                        \
+            uint32_t batch,                                                                                                          \
+            uint32_t num_rows,                                                                                                       \
+            uint32_t num_tables,                                                                                                     \
+            optimizer##_kernel_arg_t opt_karg);
+
+#define SPLIT_TBE_BWD_KERNEL_ALL_WDM(optimizer,                                 \
+                                          segment_split,                        \
+                                          emb_prec,                             \
+                                          emb_type,                             \
+                                          grad_prec,                            \
+                                          grad_type,                            \
+                                          embedding_dim,                        \
+                                          segment_prefetch,                     \
+                                          segment_unroll)                       \
+    __SPLIT_TBE_BWD_KERNEL(optimizer, 0, segment_split, emb_prec, emb_type, grad_prec, grad_type, embedding_dim, segment_prefetch, segment_unroll)  \
+    __SPLIT_TBE_BWD_KERNEL(optimizer, 1, segment_split, emb_prec, emb_type, grad_prec, grad_type, embedding_dim, segment_prefetch, segment_unroll)  \
+    __SPLIT_TBE_BWD_KERNEL(optimizer, 2, segment_split, emb_prec, emb_type, grad_prec, grad_type, embedding_dim, segment_prefetch, segment_unroll)
+
+
+#define SPLIT_TBE_BWD_KERNEL(optimizer,                               \
+                                segment_split,                        \
+                                embedding_dim)                        \
+    SPLIT_TBE_BWD_KERNEL_ALL_WDM(optimizer, segment_split, fp32, float, fp32, float, embedding_dim, 2, 8) \
+    SPLIT_TBE_BWD_KERNEL_ALL_WDM(optimizer, segment_split, fp32, float, fp16,  half, embedding_dim, 2, 8) \
+    SPLIT_TBE_BWD_KERNEL_ALL_WDM(optimizer, segment_split, fp16,  half, fp32, float, embedding_dim, 2, 8) \
+    SPLIT_TBE_BWD_KERNEL_ALL_WDM(optimizer, segment_split, fp16,  half, fp16,  half, embedding_dim, 2, 8) 
+
+// warp per row
+SPLIT_TBE_BWD_KERNEL(rowwise_adagrad, 0, 64);
+SPLIT_TBE_BWD_KERNEL(rowwise_adagrad, 0, 128);
+SPLIT_TBE_BWD_KERNEL(rowwise_adagrad, 0, 192);
+SPLIT_TBE_BWD_KERNEL(rowwise_adagrad, 0, 256);
```

Comment:
It is a minor issue, but please add a newline after `SPLIT_TBE_BWD_KERNEL(rowwise_adagrad, 0, 256);`.